Unity/Assets/YarnSpinner/Scripts/DialogueRunner.cs

```diff
@@ -664,17 +657,54 @@ void RunDialogue (string startNode = "Start")
 
         // Expose the RegisterFunction methods from dialogue.library to Unity
 
-        /// Registers a new function that returns a value, so that it can
-        /// be called from Yarn scripts.
-        public void RegisterFunction(string name, int parameterCount, ReturningFunction implementation) {
+        [Obsolete("Use AddFunction instead")]
+        public void RegisterFunction(string name, int parameterCount, ReturningFunction implementation)
+        {
+            if (dialogue.library.FunctionExists(name)) {
```

Comment:
This should be a call to AddFunction, rather than duplicating the code.

Reply:
apologies for that! resolved in commit below